### PR TITLE
Add script to filter dark_attitudes.dat by zodi brightness

### PR DESCRIPTION
**Pull request by jeanconn**
Add script to filter dark_attitudes.dat by zodi brightness

The plan is to set this to run as a cron task with the date for filtering set to ~2-3 weeks in the future from the script run time.

### CODE REVIEW

**Comment by taldcroft** [comment]
What would be the advantage in running it manually?  If it were only up to me then surely I would forget unless we had a cron job running to check the DSN schedule and send a reminder.

**Comment by jeanconn** [comment]
This is the script I used to update it manually this time (that I stuck in a PR so I wouldn't lose it).  The PR comment is from before we had the meeting and you suggested a cron task ... which hadn't occurred to me because I was confused about when dark cal attitudes were actually picked.  Comment now updated.

File: filter_atts_for_fot.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        ok.append(np.any(coord.separation(zodi_ok_coords) < (1 * u.arcsec)))
+    ok = np.array(ok)
+    cut_list = all_atts[ok]
+    return cut_list
+
+if __name__ == '__main__':
+    opt = get_options()
+    date = DateTime(opt.date)
+    cut_list = filter_atts(date)
+    cut_list.write("{}_dark_attitudes.dat".format(date.date[0:8]),
```
Comment (taldcroft) [inline filter_atts_for_fot.py]: This should be a fixed name, and ideally there should be minimal impact to FOT process.

File: filter_atts_for_fot.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    """
+    date = DateTime(date).date
+
+    Z_LOW = 0
+    Z_UP = 250
+    zodi_annotated_html = bash(
+        'perl ./attitude.pl date="{date}" type="ATT List" z_low="{z_low}" z_up="{z_up}"'.format(
+            date=date[0:8], z_low=Z_LOW, z_up=Z_UP))
+    zodi_list = Table.read(zodi_annotated_html, format='ascii.html')
+    zodi_ok = (zodi_list['Zodi'] >= Z_LOW) & (zodi_list['Zodi'] <= Z_UP)
```
Comment (taldcroft) [inline filter_atts_for_fot.py]: As always, use the standard idiom `ok` for short one-off masks.

Comment (taldcroft) [reply filter_atts_for_fot.py]: But why is this filtering needed anyway?  Doesn't the perl script do the same filtering?

File: filter_atts_for_fot.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+from Chandra.Time import DateTime
+
+
+def get_options():
+    parser = argparse.ArgumentParser(
+        description="""Make a filtered version of dark_attitudes.dat using the zodiacal brightness
+                       data from the attitude.pl CGI""")
+    parser.add_argument("--date",
+                        required=True,
+                        help="Requested date of expected dark current calibration")
```
Comment (taldcroft) [inline filter_atts_for_fot.py]: Not required.  Doc should reflect the default 2 weeks in advance thing.

File: filter_atts_for_fot.py (hunk 2 above)
Comment (jeanconn) [reply filter_atts_for_fot.py]: IIRC, the Perl script just calculates the zodiacal brightness, and the default "list" output then colorizes the lines with brightness outside the range.  To clean this up, I could either fix the Perl or take a few minutes to write the Python to  read from the zodiacal brightness lookup table directly.

File: filter_atts_for_fot.py (hunk 1 above)
Comment (jeanconn) [reply filter_atts_for_fot.py]: The cron task will update http://cxc.cfa.harvard.edu/mta/ASPECT/dark_attitudes/dark_attitudes.dat
either directly here or with another step to copy or link.

File: filter_atts_for_fot.py (hunk 2 above)
Comment (taldcroft) [reply filter_atts_for_fot.py]: OK I see.  Never mind then.

File: filter_atts_for_fot.py (hunk 1 above)
Comment (taldcroft) [reply filter_atts_for_fot.py]: OK.  We just don't want to leave hundreds of date-stamped files laying around.  All we care about is the current one.

File: filter_atts_for_fot.py (hunk 2 above)
Comment (jeanconn) [reply filter_atts_for_fot.py]: I'm slightly concerned about re-using `ok` in a space of 5 lines.

**Pull request merged**